language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ArrayTypeDescriptorNodeContext.java

```diff
@@ -93,9 +93,17 @@ public boolean onPreValidation(BallerinaCompletionContext context, ArrayTypeDesc
         (1) TypeName[<cursor>] ...
          */
         int cursor = context.getCursorPositionInTree();
-        Token openBracket = node.openBracket();
-        Token closeBracket = node.closeBracket();
+        for (ArrayDimensionNode arrayDimension : node.dimensions()) {
+            Token openBracket = arrayDimension.openBracket();
+            Token closeBracket = arrayDimension.closeBracket();
+            if (cursor <= openBracket.textRange().startOffset()) {
+                break;
+            }
+            if (cursor < closeBracket.textRange().endOffset()) {
+                return true;
+            }
```

Comment:
Can't we use the same logic we used earlier? Is there any breaking cases with it?

Reply:
Earlier `T[ ][ ][ ]` was parsing as 3 array-type-descriptor nodes like this`((T[ ])[ ])[ ]` where each array-type-descriptor node has one dimension.
Now it is flattened. So we need to loop through the array dimensions and see if the cursor belongs to one of `[ ]`.